flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,253 @@
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+
+        assertThat(fileSystemJobResultStore.hasJobResultEntry(DUMMY_JOB_RESULT_ENTRY.getJobId()))
+                .isTrue();
+    }
+
+    @Test
+    public void testGetDirtyResultsWithNoEntry() throws IOException {
+        assertThat(fileSystemJobResultStore.getDirtyResults()).isEmpty();
+    }
+
+    @Test
+    public void testGetDirtyResultsWithDirtyEntry() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+
+        assertThat(
+                        fileSystemJobResultStore.getDirtyResults().stream()
+                                .map(JobResult::getJobId)
+                                .collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
+    }
+
+    @Test
+    public void testGetDirtyResultsWithDirtyAndCleanEntry() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+
+        final JobResultEntry dirtyJobResultEntry = createDummyJobResultEntry();
+        fileSystemJobResultStore.createDirtyResult(dirtyJobResultEntry);
+
+        assertThat(fileSystemJobResultStore.getDirtyResults()).hasSize(1);
+        assertThat(getCleanResultIds()).hasSize(1);
+
+        assertThat(
+                        fileSystemJobResultStore.getDirtyResults().stream()
+                                .map(JobResult::getJobId)
+                                .collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(dirtyJobResultEntry.getJobId());
+    }
+
+    @Test
+    public void testDeleteOnCommit() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, true);
+
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+
+        File dirtyFile = new File(expectedDirtyPath(DUMMY_JOB_RESULT_ENTRY).getPath());
+        File cleanFile = new File(expectedCleanPath(DUMMY_JOB_RESULT_ENTRY).getPath());
+
+        assertThat(dirtyFile).exists().isFile().isNotEmpty();
+
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(dirtyFile).doesNotExist();
+        assertThat(cleanFile).doesNotExist();
+    }
+
+    private static JobResultEntry createDummyJobResultEntry() {
+        return new JobResultEntry(
+                new JobResult.Builder()
+                        .applicationStatus(ApplicationStatus.SUCCEEDED)
+                        .jobId(new JobID())
+                        .netRuntime(Long.MAX_VALUE)
+                        .build());
+    }
+
+    private List<JobID> getCleanResultIds() throws IOException {
```

Comment:
```suggestion
    private List<JobID> getCleanResultIdsFromFileSystem() throws IOException {
```
nit: Being more explicit in the method's name about what the method does helps reading the code later one (reduces the necessity to actually look where the information is coming from).